test/integration_test/common_test.go

```diff
@@ -9,18 +9,56 @@ import (
 
 	corev1 "github.com/libopenstorage/operator/pkg/apis/core/v1"
 	k8sutil "github.com/libopenstorage/operator/pkg/util/k8s"
+	"github.com/portworx/sched-ops/k8s/core"
 	"github.com/portworx/sched-ops/k8s/operator"
+	"github.com/sirupsen/logrus"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
 )
 
 const (
 	specDir = "./specs"
 )
 
 func TestMain(m *testing.M) {
+	if err := setup(); err != nil {
+		logrus.Errorf("Setup failed with error: %v", err)
+		os.Exit(1)
+	}
 	os.Exit(m.Run())
 }
 
+func setup() error {
+	if err := setKubeconfig(""); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func setKubeconfig(kubeConfig string) error {
+	var config *rest.Config
+	var err error
+
+	if kubeConfig == "" {
+		config = nil
+	} else {
+		config, err = clientcmd.BuildConfigFromFlags("", kubeConfig)
+		if err != nil {
+			return err
+		}
+	}
+
+	k8sOps := core.Instance()
+	k8sOps.SetConfig(config)
+
+	operatorOps := operator.Instance()
+	operatorOps.SetConfig(config)
+
+	return nil
+}
```

Comment:
You don't need to do any of this. Just using `core.Instance()` or `operator.Instance()` anywhere in the code automatically initialize the kubernetes client to the current cluster. sched-ops handles that.
You need to call SetConfig only if you want to talk to a different Kubernetes cluster from this one. Stork has to perform migration tests, that's why it needs it. We don't need it here.

The assumption here is that we will be running the test pod in the same cluster as the operator. If you want to run the test locally, you can set the KUBECONFIG variable and it should talk to that k8s cluster.

Reply:
Good to know, removed.